src/main/java/avd/inf/jdm/rentmycar/domain/User.java

```diff
@@ -23,13 +24,30 @@ public class User extends Account {
     private LocalDate dateOfBirth;
     private int bonusPoints;
 
+    private boolean isAdult;
 
     public User(String firstName, String lastName, String password, LocalDate dateOfBirth, String email, int bonusPoints) {
         super(email, password);
         this.firstName = firstName;
         this.lastName = lastName;
         this.dateOfBirth = dateOfBirth;
         this.bonusPoints = bonusPoints;
+        this.isAdult = isAdult();
     }
 
+    public int calculateAge() {
+        LocalDate curDate = LocalDate.now();
+        if ((dateOfBirth != null) && (curDate != null)) {
+            return Period.between(dateOfBirth, curDate).getYears();
+        } else {
+            return 0;
+        }
+    }
+
+    public boolean isAdult() {
+        int userAge = calculateAge();
```

Comment:
I think you can shorten this method to 
`public boolean isAdult() {
  return calculateAge() >= 18
}`

Reply:
nice oneliner 👍  thanks